### DISCUSSION TITLE
Loading "external" jars on startup. (Plugin System)

### DISCUSSION THREAD

**Discussion by ckosmowski** [quarkusio/quarkus#39103] [Q&A]
For something like a plugin system i am searching for a way to:

Option a) Load external jar files on startup from a specified folder that contain implementations of an SPI Interface that i can then use inside my quarkus application.

Option b)  Load external jar files on startup from a specified folder that contain CDI annotated Beans that are loaded by quarkus and made available via the bean context.

The background is that i need to deploy a base quarkus application and augment it with stakeholder specific Connector Implementations without needing to build a separate version of the application for each stakeholder.

Disclaimer: I of course googled this, but i couldn't find a solution or a workaround that helps me achieve this. I understand that generally quarkus determines at build time the classes that it loads. The least i am hoping to find here is a "definitive not possible" so that i can let it go :-) The best would be any kind of solution advice.

**Comment by stuartwdouglas** [comment]
The closest you can get is a mutable jar with the user providers directory:

https://quarkus.io/guides/all-config#quarkus-core_quarkus-package-user-providers-directory

Basically add your plugin jar files to this folder and re-augment as specified here: https://quarkus.io/guides/reaugmentation

**Reply by BGhass** [reply]
Hello,

I tested your solution. It works fine in Dev Mode.

But, how can I use it in a native executable (re-augment app before generating native executable)?
